backend/src/middleware/isAuth.js

```diff
@@ -1,15 +1,33 @@
 const jwt = require("express-jwt");
 
 const getTokenFromHeader = (req) => {
-  if (
-    req.headers.authorization &&
-    req.headers.authorization.split(" ")[0] === "Bearer"
-  )
-    return req.headers.authorization.split(" ")[1];
+
+  const authHeader = req.get('Authorization');
+
+  if (!authHeader) {
+    const error = new Error('Not Authenticated');
+    error.statusCode = 401;
+    throw error;
+  }
+
+  const token = authHeader.split(" ")[1];
+
+  if (!token) {
+    const error = new Error('No Bearer Token');
+    error.statusCode = 500;
+    throw error;
+  }
+
+  return token;
```

Comment:
None of this is needed lol but I just needed the errors to be thrown so I could get a handle on testing. Guess it doesn't hurt to have though.

Reply:
Would throwing the error crash the application? That could mean bad requests could bring the whole thing down